control-plane/agents/core/src/core/reconciler/mod.rs

```diff
@@ -48,7 +48,10 @@ impl ReconcilerControl {
     }
 
     /// Send an event signal to the poller's main loop
+    /// todo: don't requeque duplicate events
     pub(crate) async fn notify(&self, event: PollEvent) {
-        self.event_channel.send(event).await.ok();
+        if let Err(error) = self.event_channel.try_send(event) {
+            tracing::error!(error=?error, "Failed to send event to reconcile worker");
```

Comment:
If it is legitimate that this can happen, should we emit a warning rather than an error?

Reply:
Yep good point